noir_stdlib/src/meta/expr.nr

```diff
@@ -115,6 +120,7 @@ impl Expr {
     fn mutate<Env>(self, f: fn[Env](Expr) -> Option<Expr>) -> Expr {
         // docs:end:mutate
         let result = mutate_array(self, f);
+        let result = result.or_else(|| mutate_assert(self, f));
```

Comment:
Do you mind renaming `mutate` to `modify`?
Reasoning being that this method does not mutate the inner expression

Reply:
Hm, but it does mutate the inner expressions, or not?

Reply:
No, it returns a new expression

Reply:
For all of them, I mentioned this on the function body PR but forgot about it before I approved

Reply:
Done!